tools/cassandra_delete_range/internal/cass/cass.go

```diff
@@ -0,0 +1,566 @@
+		log.Printf("Total traversed rows: %d\n\n", rowsCount)
+		totalErrors += errCount
+		totalRows += rowsCount
+		deleteCount, errCount = c.performDeleteQueries(&info, columnSettings{UseBlob: true, UseSeq: false})
+		totalErrors += errCount
+		totalDeletes += deleteCount
+	}
+
+	// diff queries
+	if !c.settings.SkipDiffTable {
+		log.Println("Generating delete queries for diff table")
+		info = c.prepareSimpleDeleteQueries(rangeFrom, rangeTo,
+			"DELETE FROM diff WHERE seq = ?")
+		log.Printf("Total delete queries: %d\n\n", len(info.Data))
+		deleteCount, errCount = c.performDeleteQueries(&info, columnSettings{UseBlob: true, UseSeq: true})
+		totalErrors += errCount
+		totalDeletes += deleteCount
+	}
+
+	// ledger_transactions queries
+	if !c.settings.SkipLedgerTransactionsTable {
+		log.Println("Generating delete queries for ledger_transactions table")
+		info = c.prepareSimpleDeleteQueries(rangeFrom, rangeTo,
+			"DELETE FROM ledger_transactions WHERE ledger_sequence = ?")
+		log.Printf("Total delete queries: %d\n\n", len(info.Data))
+		deleteCount, errCount = c.performDeleteQueries(&info, columnSettings{UseBlob: false, UseSeq: true})
+		totalErrors += errCount
+		totalDeletes += deleteCount
+	}
+
+	// ledgers queries
+	if !c.settings.SkipLedgersTable {
+		log.Println("Generating delete queries for ledgers table")
+
+		info = c.prepareSimpleDeleteQueries(rangeFrom, rangeTo,
+			"DELETE FROM ledgers WHERE sequence = ?")
+		log.Printf("Total delete queries: %d\n\n", len(info.Data))
+		deleteCount, errCount = c.performDeleteQueries(&info, columnSettings{UseBlob: false, UseSeq: true})
+		totalErrors += errCount
+		totalDeletes += deleteCount
+	}
+
+	// TODO: tbd what to do with account_tx as it got tuple for seq_idx
+	// TODO: also, whether we need to take care of nft tables and other stuff like that
+
+	if !c.settings.SkipWriteLatestLedger {
+		var (
+			first maybe.Maybe[uint64]
+			last  maybe.Maybe[uint64]
+		)
+
+		if fromLedgerIdx.HasValue() {
+			last = maybe.Set(fromLedgerIdx.Value() - 1)
+		}
+
+		if toLedgerIdx.HasValue() {
+			first = maybe.Set(toLedgerIdx.Value() + 1)
+		}
+
+		if err := c.updateLedgerRange(first, last); err != nil {
+			log.Printf("ERROR failed updating ledger range: %s\n", err)
+			return err
+		}
+	}
+
+	log.Printf("TOTAL ERRORS: %d\n", totalErrors)
+	log.Printf("TOTAL ROWS TRAVERSED: %d\n", totalRows)
+	log.Printf("TOTAL DELETES: %d\n\n", totalDeletes)
+
+	log.Printf("Completed deletion for %s -> %s\n\n", fromStr, toStr)
+
+	return nil
+}
+
+func (c *ClioCass) prepareSimpleDeleteQueries(
+	fromLedgerIdx uint64,
+	toLedgerIdx uint64,
+	deleteQueryTemplate string,
+) deleteInfo {
+	var info = deleteInfo{Query: deleteQueryTemplate}
+
+	// Note: we deliberately add 1 extra ledger to make sure we delete any data Clio might have written
+	// if it crashed or was stopped in the middle of writing just before it wrote ledger_range.
+	for i := fromLedgerIdx; i <= toLedgerIdx+1; i++ {
```

Comment:
It seems like `toLedgerId+1` is not okay for **delete_before** case.
if delete_before  parameter is 1000, then toLedgerIdx will be 999. In this for loop, it will delete 1000.